students/drovdahl/lesson08/circle.py

```diff
@@ -0,0 +1,74 @@
+#! /usr/local/bin/python3
+'''
+Class that does the following:
+    Compute the circle's area
+    Print the circle and get something nice
+    Add two circles together
+    Compare two circles to see which is bigger
+    Compare two circles to see if they are equal
+    Put circles in a list and sort them
+
+Use the following techniques
+    properties
+    classmethod
+    special methods
+'''
+import math
+
+
+class Circle(object):
+    def __init__(self, radius):
+        self._radius = radius
+        self.diameter = radius * 2
+
+    @property
+    def radius(self):
+        return self.diameter / 2
+
+    @property
+    def diameter(self):
+        return self._radius * 2
+
+    @diameter.setter
+    def diameter(self, val):
+        self._radius = val / 2
```

Comment:
Since you keep track of diameter separately (which you probably shouldn't, because the diameter can always be computed from the radius), you need to update `self.diameter` as well (`self.diameter = val`). Otherwise, subsequent calls to various methods will give incorrect values.

Reply:
OK, I 'think' I understand what you're saying here.  If you could give me an example of how a subsequent call to another method would fail given this code, that would be really helpful.

Reply:
looking now, if you comment out the 

``self.diameter = ``

line it will all work -- except being able to set the radius.

So I'd change self._radius to plain old self.radius, and get rid of the radius property.

Reply:
This is addressed in my next PR